latitudesh/terraform-provider-latitudesh
# fix: server reinstall

#### What does this PR do?
This PR aims to avoid cases where users could unintentionally trigger a server reinstall.

We verify if a reinstall operation will take place at runtime inside the Update function, which makes it hard to warn the user before the reinstall occurs, and Terraform only reports an in-place update.

#### Description of Task to be completed?
Create an `allow_reinstall` variable that will be used to let Terraform trigger the reinstall. This variable will be false by default.
Make general adjustments to the reinstall process to avoid errors.

#### How should this be manually tested?
to test the provider manually, you can follow [Notion - Developing the Terraform Provider Locally](https://www.notion.so/Developing-the-Terraform-Provider-Locally-6682fe3df39d4003b519301e630f51ef?pvs=27&qid=1%3A8c80df71-563c-4c6a-905d-be4a225ec805%3A0)

To verify:
- Creating a server with a tag will trigger a subsequent update. This update should not reinstall the server even if the triggering attributes (ssh_keys, user_data, raid, or ipxe_url) are given.
- Changing the allow_reinstall variable itself should not trigger a reinstall.
